src/components/EditTripModal.tsx

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import { Trip } from "./TripList";
+import { doc, updateDoc } from "firebase/firestore";
+import { db } from "../firebase";
+
+import { useToast } from "../contexts/ToastContext";
+import Button from "./Button";
+import classes from "../styles/EditTripModal.module.scss";
+import DateRangePicker from "./DateRangePicker";
+import { useDate } from "../contexts/DateContext";
+
+interface EditTripModalProps {
+  trip: Trip;
+  tripId: string;
+  onClose: () => void;
+}
+
+const EditTripModal: React.FC<EditTripModalProps> = ({
+  trip,
+  tripId,
+  onClose,
+}) => {
+  const { startDate, endDate, cancelSelectedDates } = useDate();
+
+  const toLocalDateString = (date: Date) => {
+    const localDate = new Date(date);
+
+    localDate.setMinutes(
+      localDate.getMinutes() - localDate.getTimezoneOffset()
+    );
+    return localDate.toISOString().split("T")[0];
+  };
+
+  const addToast = useToast();
+
+  const handleOnClose = () => {
+    onClose();
+    cancelSelectedDates();
+  };
+
+  const handleSave = async () => {
+    try {
+      if (!startDate || !endDate) {
+        addToast("Start date and end date cannot be empty.", "error");
+        return;
+      }
```

Comment:
Lets disable the save button if here is no startData and endDate. So there is no need for this error in the first place